frontend/hub/access/roles/RolePage/RoleForm.tsx

```diff
@@ -0,0 +1,176 @@
+import { useTranslation } from 'react-i18next';
+import {
+  LoadingPage,
+  PageFormSubmitHandler,
+  PageFormTextInput,
+  PageHeader,
+  PageLayout,
+  useGetPageUrl,
+} from '../../../../../framework';
+import { PageFormSection } from '../../../../../framework/PageForm/Utils/PageFormSection';
+import { Role } from '../Role';
+import { PageFormRolePermissionsSelect } from '../components/PageFormPermissionsSelect';
+import { HubRoute } from '../../../main/HubRoutes';
+import { HubPageForm } from '../../../common/HubPageForm';
+import { useNavigate, useParams } from 'react-router-dom';
+import { useGet } from '../../../../common/crud/useGet';
+import { pulpAPI } from '../../../common/api/formatPath';
+import { PermissionCategory, usePermissionCategories } from '../components/RolePermissions';
+import { usePostRequest } from '../../../../common/crud/usePostRequest';
+import { useIsValidRoleName } from '../hooks/useIsValidRoleName';
+import { useIsValidRoleDescription } from '../hooks/useIsValidRoleDescription';
+import { usePatchRequest } from '../../../../common/crud/usePatchRequest';
+import { parsePulpIDFromURL } from '../../../common/api/hub-api-utils';
+import { HubError } from '../../../common/HubError';
+import { PulpItemsResponse } from '../../../common/useHubView';
+
+export interface RoleInput extends Omit<Role, 'pulp_href' | 'pulp_created' | 'locked'> {
+  permissionCategories: PermissionCategory[];
+}
+
+export type RoleRequestBody = Omit<Role, 'pulp_href' | 'pulp_created' | 'locked'>;
+
+export function CreateRole() {
+  const { t } = useTranslation();
+  const getPageUrl = useGetPageUrl();
+  const navigate = useNavigate();
+  const postRequest = usePostRequest<RoleRequestBody>();
+  const onSubmit: PageFormSubmitHandler<RoleInput> = async (values) => {
+    const { name, description, permissionCategories } = values;
+    const permissions = getSelectedPermissions(permissionCategories);
+    await postRequest(pulpAPI`/roles/`, {
+      name,
+      description,
+      permissions,
+    });
+    navigate(-1);
```

Comment:
I just noticed that one now. I believe we are redirecting to the details page after creating resources in general.

Reply:
updated 👍